File: 0949-58-2559-count-vowel-strings-in-ranges/0949-58-2559-count-vowel-strings-in-ranges.java

class Solution {
    public  boolean isVowel(char ch) {
            return ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u';
    }
    public int[] vowelStrings(String[] words, int[][] queries) {
        int n = words.length;
        int[] pre = new int[n];

       

        for (int i = 0; i < n; i++) {
            if (isVowel(words[i].charAt(0)) && isVowel(words[i].charAt(words[i].length() - 1))) {
                pre[i] = 1;
            }
            if (i > 0) {
                pre[i] += pre[i - 1];
            }
        }

        int[] ans = new int[queries.length];
        for (int i = 0; i < queries.length; i++) {
            int l = queries[i][0], r = queries[i][1];
            ans[i] = pre[r] - (l > 0 ? pre[l - 1] : 0);
        }

        return ans;
    }
}

File: 0949-58-2559-count-vowel-strings-in-ranges/0949-58-2559-count-vowel-strings-in-ranges.py
class Solution:
    def vowelStrings(self, words: list[str], queries: list[list[int]]) -> list[int]:
        def is_vowel(ch: str) -> bool:
            return ch in 'aeiou'

        n = len(words)
        pre = [0] * n

        for i in range(n):
            pre[i] = is_vowel(words[i][0]) and is_vowel(words[i][-1])
            if i > 0:
                pre[i] += pre[i - 1]

        ans = []
        for l, r in queries:
            ans.append(pre[r] - (pre[l - 1] if l > 0 else 0))
        
        return ans
